src/Airy/airy.jl

```diff
@@ -231,37 +233,42 @@ function airybiprimex(x::Float64)
     q = evalpoly(invx3, (-1.5707963267948966, 0.15510250188621483, 0.4982993247266722, 5.515384839161109, 129.24738229725767, 5209.103946324185, 321269.61208650155, 2.812618811215662e7, 3.3166403972012258e9, 5.0676100258903735e11, 9.738286496397669e13, 2.298637212441062e16))
     q1 = evalpoly(invx3, (0.22907446432425577, 0.22511404787652015, 1.4803642438754887, 24.70432792540913, 773.390007496322, 38999.21950723391, 2.8878225227454924e6, 2.950515261265541e8, 3.97712331943799e10, 6.837383921993536e12, 1.460066704564067e15, 3.7912939312807334e17))
     
-    sqrinvx3 = sqrt(invx3)
-    xsqr = sqrt(sqrt(x))
-    bi = muladd(sqrinvx3, p1, p) / (xsqr * PIPOW3O2(T))
-    bip = xsqr * muladd(sqrinvx3, q1, q) / PIPOW3O2(T)
+    xsqr = sqrt(x)
+    xsqrx = 1 / (x * xsqr)
+    xsqr = sqrt(xsqr)
+    bi = muladd(xsqrx, p1, p) / (xsqr * PIPOW3O2(T))
+    bip = xsqr * muladd(xsqrx, q1, q) / PIPOW3O2(T)
     return 2*bi, -2*bip
 end
 
 # scaled factor is exp(-|2/3 Re(z^(3/2))|)
 # for negative arguments the scaled Bi versions are equivalent airybi = airybix
 function airybi_large_neg_arg(x::T) where T <: Float64
-    x3 = x*x*x
-    invx3 = 1 / x3
+    invx3 = 1 / x^3
     p = evalpoly(invx3, (1.5707963267948966, 0.13124057851910487, 0.4584353787485384, 5.217255928936184, 123.97197893818594, 5038.313653002081, 312467.7049060495, 2.746439545069411e7, 3.2482560591146026e9, 4.97462635569055e11, 9.57732308323407e13, 2.2640712393216476e16))
     p1 = evalpoly(invx3, (0.1636246173744684, 0.20141783231057064, 1.3848568733028765, 23.555289417250567, 745.2667344964557, 37835.063701047824, 2.8147130917899106e6, 2.8856687720069575e8, 3.8998976239149216e10, 6.718472897263214e12, 1.4370735281142392e1, 3.7367429394637446e17))
     q = evalpoly(invx3, (-1.5707963267948966, 0.15510250188621483, 0.4982993247266722, 5.515384839161109, 129.24738229725767, 5209.103946324185, 321269.61208650155, 2.812618811215662e7, 3.3166403972012258e9, 5.0676100258903735e11, 9.738286496397669e13, 2.298637212441062e16))
     q1 = evalpoly(invx3, (0.22907446432425577, 0.22511404787652015, 1.4803642438754887, 24.70432792540913, 773.390007496322, 38999.21950723391, 2.8878225227454924e6, 2.950515261265541e8, 3.97712331943799e10, 6.837383921993536e12, 1.460066704564067e15, 3.7912939312807334e17))
     
-    sqrinvx3 = im*sqrt(abs(invx3))
-    xsqr = sqrt(abs(x))
-    c = cis(2 * x * xsqr / 3)
-    xsqr = complex(sqrt(xsqr/2), sqrt(xsqr/2)) # sqrt(im*xsqr)
-    bi = muladd(sqrinvx3, p1, p) / (xsqr * PIPOW3O2(T))
-    bip = xsqr * muladd(sqrinvx3, q1, q) / PIPOW3O2(T)
-    return real(2 * bi * c), real(-2 * bip * c)
+    xabs = abs(x)
+    xsqr = sqrt(xabs)
+    xsqrx = xabs * xsqr
+    scale = cis(2 * x * xsqr / 3)
+    b = muladd(im / xsqrx, p1, p) * scale
+    d = muladd(im / xsqrx, q1, q) * scale
+
+    xsqr = sqrt(xsqr * 0.5)
+    cxsqr = complex(xsqr, xsqr) # sqrt(im*xsqr)
+
+    bi = 2*real(b / (cxsqr * PIPOW3O2(T)))
+    bip = -2*real(cxsqr * d / PIPOW3O2(T))
```

Comment:
`-2 * cxsqr / PIPOW3O2(T) * real(d)`?

Reply:
I think there's a potential for speed up here by unrolling the real parts (ignore the imaginary parts) but `b`, `d`, and `cxsqr` are complex. Since `cxsqr` has identical real and imaginary parts we can probably just simplify `bip = -2 / PIPOW3O2(T)  * (real(d) - imag(d)) * real(cxsqr)` but also change `cxsqr` to just be one part. I'm wondering if we can just completely unroll all these imaginary parts away. This section is also prone to cancellation.... Ill work on this too

Reply:
Yep this simplifies down to (which avoids complex divisions)..
```julia
b = muladd(im / xsqrx, p1, p) * scale
d = muladd(im / xsqrx, q1, q) * scale

xsqr = sqrt(xsqr * 0.5)
bi = (real(b) + imag(b)) / (xsqr * PIPOW3O2(T))
bip = -2 / PIPOW3O2(T) * (real(d) - imag(d)) * xsqr
return bi, bip
```

Reply:
you can pull out the `*.5` from the `sqrt` but other than that, this looks a lot better. Complex division really sucks.

Reply:
Unrolling all of this. I haven't checked it it is faster but probably (if we do `sincos` instead). It becomes...
```julia
    xabs = abs(x)
    xsqr = sqrt(xabs)
    xsqrx = xabs * xsqr
    z = 2 * x * xsqr / 3
    b = p * (sin(z) + cos(z)) + p1 * (cos(z) - sin(z)) / xsqrx
    d = q * (cos(z) - sin(z)) - q1 * (cos(z) + sin(z)) / xsqrx

    xsqr = sqrt(xsqr * 0.5)
    bi = b / (xsqr * PIPOW3O2(T))
    bip = -2 / PIPOW3O2(T) * d * xsqr
    return bi, bip
```
The `cos(z) - sin(z)` is prone to cancellation though so need to work on that more and shouldn't use this directly.

Reply:
that's just `sqrt(2)*sin(x-pi/4)`. I really need to write you a good function for `sin` with quarter-pi offsets.

Reply:
Yes! It pops up everywhere. I remember seeing this in openlibm https://github.com/JuliaMath/openlibm/blob/1d2c5e3bf52e050c994510ede91e0d2f0717173a/src/e_j0.c#L29-L37... which offered that work around.

Reply:
Recent commit has avoided any complex arithmetic here. Though now we must compute (cos(x) +- sin(x)). I'm tempted to have a separate issue for computing the trig functions with quarter-pi offsets as they pop up in several functions. We can update these when we get a good functions. This implementation is fairly strictly superior to the current version (more accurate and much faster) so I don't think we have to wait for such an implementation before merging.

TODO: 1. Use similar approach for the airyai function to avoid complex functions.
              2. Need to have a separate infinite branch for negative arguments to avoid computing `sin` at negative infinity. Overall function goes to zero.
              3. Add docs.
              4. Define a better test structure for this function.